src/test/components/AppLocalizationProvider.test.js

```diff
@@ -66,4 +89,22 @@ describe('AppLocalizationProvider', () => {
       getByTestId('content-test-AppLocalizationProvider')
     ).toBeInTheDocument();
   });
+
+  it('fetches the en-US FTL strings and renders them', async () => {
+    const { store, translatedText } = setup();
+    const { getByText } = render(
+      <Provider store={store}>
+        <AppLocalizationProvider>
+          <Localized id="test-id">
+            <span>Fallback String</span>
+          </Localized>
+        </AppLocalizationProvider>
+      </Provider>
+    );
+    await waitUntilState(
+      store,
+      state => getL10nFetchingPhase(state) === 'done-fetching'
+    );
+    expect(getByText(translatedText)).toBeTruthy();
```

Comment:
Have you tried using:
```js
expect(await findByText(translatedText)).toBeTruthy();
```
instead of waiting for the state?

I believe this is simpler, and less coupled to the internal implementation, and therefore more future-proof, more solid, and less maintenance needed!

Reply:
Some more explanation: `findByText` will wait (for some time) until the text is available.
You can see [all queries on the documentation for the testing library](https://testing-library.com/docs/queries/about#types-of-queries).

Reply:
This works! I guess it looks cleaner now. Thanks!